staging/src/k8s.io/client-go/util/workqueue/parallelizer.go

```diff
@@ -50,3 +51,37 @@ func Parallelize(workers, pieces int, doWorkPiece DoWorkPieceFunc) {
 	}
 	wg.Wait()
 }
+
+// ParallelizeUntil is a very simple framework that allows for parallelizing
+// N independent pieces of work until the context done or canceled.
+func ParallelizeUntil(workers, pieces int, doWorkPiece DoWorkPieceFunc, ctx context.Context) {
```

Comment:
isn't ctx always the first arg by Golang convention?

Reply:
Yes, indeed. I can change it as the first ctx.